### ISSUE TITLE
Updating DyNet configuration parameters

### ISSUE THREAD

**Issue by danielhers** [clab/dynet#1071]
Is there a way to update the parameters after DyNet is initialized? For example, in a hyperparameter-tuning script, I'd like to change `weight_decay` for each new sample of hyperparameters.

I don't understand the difference between `DynetParams` and `dynet_config`. The [documentation](http://dynet.readthedocs.io/en/latest/python_ref.html) says I can use either, but `dynet_config` must be used before `import dynet`. However, it seems like I can't use `DynetParams` after `import dynet` either, because once I call `dyparams.init()` I get

    WARNING: Attempting to initialize dynet twice. Ignoring duplicate initialization.

**Comment by neubig** [comment]
As far as I know there's no way to do this. I think the DyNet command line parameters were basically meant to be ones that shouldn't be changed after initialization, but it might be worth adding functions to change some of the relevant ones like weight decay.

**Comment by danielhers** [comment]
Thanks. I opened an issue about weight decay. Regarding `DynetParams` and `dynet_config`, I think there's a documentation issue - it was not clear to me from the documentation that neither can be updated, and I still don't understand why both mechanisms exist.
